### PR TITLE
Typo fixes on JS page [ci skip]

### PR DESCRIPTION
**Pull request by CodingItWrong**
- Makes capitalization of bundler tool names consistent with official docs (lowercase for esbuild, capitalized for others)
- Fixes an instance of a double-hyphen command flag showing as an em-dash (or en-dash, not sure)
- Fixes a sentence that does not read correctly since it was edited when adding Bun

### CODE REVIEW

File: guides/source/working_with_javascript_in_rails.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ development.
 
 ### Installing a JavaScript Runtime
 
-If you are using a esbuild, rollup.js or Webpack, to bundle your JavaScript in
+If you are using esbuild, Rollup.js or Webpack to bundle your JavaScript in
```
Comment (skipkayhil) [inline guides/source/working_with_javascript_in_rails.md:90]:
```suggestion
If you are using esbuild, Rollup.js, or Webpack to bundle your JavaScript in
```

File: guides/source/working_with_javascript_in_rails.md [hunk 2]
```diff
@@ -103,11 +103,11 @@ $ bun --version
 The version of your Bun runtime should be printed out. If it says something
 like `1.0.0`, Bun has been installed correctly.
 
-If not, you may need to reinstall bun in the current directory or restart your terminal.
+If not, you may need to reinstall Bun in the current directory or restart your terminal.
 
 #### Installing Node.js and Yarn
 
-If you are using esbuild, rollup.js or Webpack you will need Node.js and Yarn.
+If you are using esbuild, Rollup.js or Webpack you will need Node.js and Yarn.
```
Comment (skipkayhil) [inline guides/source/working_with_javascript_in_rails.md:110]:
```suggestion
If you are using esbuild, Rollup.js, or Webpack you will need Node.js and Yarn.
```

**Review by skipkayhil — COMMENTED** [review]
Two small changes, then can you squash your commits into one?

**Comment by CodingItWrong** [comment]
@skipkayhil Thanks! I've applied your fixes, squashed all the commits into one, and force-pushed it up

**Comment by skipkayhil** [comment]
Thanks for the PR!

**Pull request merged**